src/main/cpp/agent.cpp

```diff
@@ -88,7 +88,8 @@ void JNICALL OnVMDeath(jvmtiEnv *jvmti_env, JNIEnv *jni_env) {
     IMPLICITLY_USE(jvmti_env);
     IMPLICITLY_USE(jni_env);
 
-    prof->stop();
+    if (prof->isRunning())
```

Comment:
It fixes a really minor issue: if profiler was stopped before VM Death event, message 'Stopping sampling' will be printed twice, which is quite confusing. Alternatively I can move this check inside `Profiler::stop`.